cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/resource/volume/AwsVolumeIopsAndThroughputCalculator.java

```diff
@@ -0,0 +1,46 @@
+package com.sequenceiq.cloudbreak.cloud.aws.common.resource.volume;
+
+import org.springframework.stereotype.Component;
+
+/**
+ * This class calculates the IOPS and throughput values for GP3 volumes to be equivalent to the IOPS and throughput values for GP2 volumes in the same size.
+ * For other volume types it returns with null
+ * <p>
+ * Calcualtion is based on the followings: https://aws.amazon.com/blogs/storage/migrate-your-amazon-ebs-volumes-from-gp2-to-gp3-and-save-up-to-20-on-costs/
+ */
+@Component
+public class AwsVolumeIopsAndThroughputCalculator {
+
+    private static final int MIN_GP3_IOPS = 3000;
+
+    private static final int MAX_GP3_IOPS = 16000;
+
+    // It is a magic constant from the AWS documentation
+    private static final int IOPS_MULTIPLIER = 3;
+
+    private static final int MIN_GP3_THROUGHPUT = 125;
+
+    private static final int GP3_SIZE_500GB = 500;
+
+    private static final int GP3_THROUGHPUT_OVER_500GB = 250;
+
+    public Integer getIops(String type, Integer size) {
+        if (!"gp3".equalsIgnoreCase(type)) {
+            return null;
```

Comment:
AwsVolumeIopsAndThroughputCalculator does not tell me it only supports gp3 calculations.
Wouldn't it be better to throw an error for unsupportedDiskType rather than returning null?
or does aws ignore `.withIops(null)`

Reply:
null is the default value, and AWS SDK treats it as ignore (not-set) or default to minimum, depending on the volume type.

The way it is implemented is quite concise,  and if EbsBlockDevice.java would have supported something like "NOT_SET" then I would have returned it. 

Since EbsBlockDevice.java uses a kind of builder syntax like .withSomething(..), it would make things a bit more complex, but I could do something like:

```java
EbsBlockDevice ebsBlockDevice = new EbsBlockDevice()
                .withDeleteOnTermination(true)
                .withVolumeType(volumeType)
                .withIops(awsVolumeIopsAndThroughputCalculator.getIops(volumeType, rootVolumeSize))
                .withThroughput(awsVolumeIopsAndThroughputCalculator.getThroughput(volumeType, rootVolumeSize))
                .withVolumeSize(rootVolumeSize);

if(awsVolumeIopsAndThroughputCalculator.isIOPSCalculationSupported(volumeType)) {
     ebsBlockDevice = 
                ebsBlockDevice.withIops(awsVolumeIopsAndThroughputCalculator.getIops(volumeType, rootVolumeSize));
}


if(awsVolumeIopsAndThroughputCalculator.isThrougputCalculationSupported(volumeType)) {
     ebsBlockDevice = 
                ebsBlockDevice.withThroughput(awsVolumeIopsAndThroughputCalculator.getThroughput(volumeType, rootVolumeSize));
}

```


Probably in this case I would split AwsVolumeIopsAndThroughputCalculator to AwsVolumeIopsCalculator and AwsVolumeThroughputCalculator.

Reply:
I have refactored a bit, but with a lot of if isSupported it looked quite ugly. So I have decided to split the calculator to two classes, and added extra test to cover that the EbsBlockDevice.getIops and EbsBlockDevice.getThroughput are defaulting to null. If AWS SDK behavior is changed then those tests will fail.